From f76afb3cff831f475c275c88124320e56a9d07c8 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Thu, 23 Dec 2021 17:36:29 -0700
Subject: [PATCH 09/13] HACK: Drivers: hv: vmbus: Workaround clang bug

 drivers/hv/vmbus_drv.c:2082:29: error: shift count >= width of type [-Werror,-Wshift-count-overflow]
 static u64 vmbus_dma_mask = DMA_BIT_MASK(64);
                             ^~~~~~~~~~~~~~~~
 ./include/linux/dma-mapping.h:76:54: note: expanded from macro 'DMA_BIT_MASK'
 #define DMA_BIT_MASK(n) (((n) == 64) ? ~0ULL : ((1ULL<<(n))-1))
                                                      ^ ~~~
 1 error generated.

Link: https://lore.kernel.org/r/YcC1CobR%2Fn0tJhdV@archlinux-ax161/
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/hv/vmbus_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hv/vmbus_drv.c b/drivers/hv/vmbus_drv.c
index 17bf55fe3169..2376ee484362 100644
--- a/drivers/hv/vmbus_drv.c
+++ b/drivers/hv/vmbus_drv.c
@@ -2079,7 +2079,7 @@ struct hv_device *vmbus_device_create(const guid_t *type,
 	return child_device_obj;
 }
 
-static u64 vmbus_dma_mask = DMA_BIT_MASK(64);
+static u64 vmbus_dma_mask = ~0ULL;
 /*
  * vmbus_device_register - Register the child device
  */
-- 
2.35.1

